### PR TITLE
Pyzhou/tdi 47234 t xslt regression

### PR DESCRIPTION
**Pull request by pyzhou-talend**
**What is the current behavior?** (You can also link to an open issue here)
https://jira.talendforge.org/browse/TDI-47234

**What is the new behavior?**

**Please check if the PR fulfills these requirements**

- [ ] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: main/plugins/org.talend.repository/src/main/java/org/talend/repository/model/migration/XSLTCheckCacheFileInMemoryTask.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+                    ComponentUtilities.addNodeProperty(node, "CACHE_SOURCE", "CHECK");//$NON-NLS-1$ //$NON-NLS-2$
+                    ComponentUtilities.getNodeProperty(node, "CACHE_SOURCE")
+                            .setValue("true");//$NON-NLS-1$ //$NON-NLS-2$
+                }
+            }
+
+        };
+
+        for (String name : compNames) {
+            IComponentFilter filter = new NameComponentFilter(name);
```
Comment (jzhao-talend) [inline main/plugins/org.talend.repository/src/main/java/org/talend/repository/model/migration/XSLTCheckCacheFileInMemoryTask.java:47]: please refer to the change of TUP https://github.com/Talend/tdi-studio-se/commit/eff697cac2eb73c6e781186618832a6ed2ea2cfb#diff-1e40c814b68a78e4c70adbaa74c68f46655088d73800692c598532b24161fb37
modified ---> true    return ExecutionResult.SUCCESS_NO_ALERT
exception ---> ExecutionResult.FAILURE
else ---> ExecutionResult.NOTHING_TO_DO
this is used to provide migration repport.

File: main/plugins/org.talend.repository/src/main/java/org/talend/repository/model/migration/XSLTCheckCacheFileInMemoryTask.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+            try {
+                ModifyComponentsAction.searchAndModify(item, processType, filter, Arrays
```
Comment (jzhao-talend) [inline main/plugins/org.talend.repository/src/main/java/org/talend/repository/model/migration/XSLTCheckCacheFileInMemoryTask.java:50]: seems better:
```
        boolean modified = false;
        for (String name : compNames) {
            IComponentFilter filter = new NameComponentFilter(name);

            try {
                modified |= ModifyComponentsAction.searchAndModify(item, processType, filter, Arrays.asList(action));
            } catch (PersistenceException e) {
                ExceptionHandler.process(e);
                return ExecutionResult.FAILURE;
            }
        }

        if (modified) {
            return ExecutionResult.SUCCESS_NO_ALERT;
        } else {
            return ExecutionResult.NOTHING_TO_DO;
        }
```

Comment (pyzhou-talend) [reply main/plugins/org.talend.repository/src/main/java/org/talend/repository/model/migration/XSLTCheckCacheFileInMemoryTask.java:50]: Yes. I will change that

**Comment by pyzhou-talend** [comment]
Hi @jzhao-talend Correct the Migration according to your comment, please help to review again.

File: main/plugins/org.talend.repository/src/main/java/org/talend/repository/model/migration/XSLTCheckCacheFileInMemoryTask.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+        };
+
+        boolean modified = false;
+
+        for (String name : compNames) {
+            IComponentFilter filter = new NameComponentFilter(name);
+
+            try {
+                modified = ModifyComponentsAction.searchAndModify(item, processType, filter, Arrays.asList(action));
```
Comment (jzhao-talend) [inline main/plugins/org.talend.repository/src/main/java/org/talend/repository/model/migration/XSLTCheckCacheFileInMemoryTask.java:51]: the modified value would be determined by the last item.
this component list only has one component. this should be ok.
need to be careful when the list has more than one component.

**Review by jzhao-talend — APPROVED** [review]
LGTM

File: main/plugins/org.talend.repository/src/main/java/org/talend/repository/model/migration/XSLTCheckCacheFileInMemoryTask.java (hunk 3 above)
Comment (pyzhou-talend) [reply main/plugins/org.talend.repository/src/main/java/org/talend/repository/model/migration/XSLTCheckCacheFileInMemoryTask.java:51]: OK. I will take care next time.

**Pull request merged**